File: src/main/node/net/ecma/Library/Examples/ICTS/stage/infowall2.mjs
'use strict';
var X3D = require('./x3d.mjs');
try {
	var fs = require('fs');
} catch (e) {
console.log("Problems loading fs. On browser?",e);
}
var SFString = require('./x3d.mjs');
var SFNode = require('./x3d.mjs');
var head = require('./x3d.mjs');
var MFNode = require('./x3d.mjs');
var meta = require('./x3d.mjs');
var Scene = require('./x3d.mjs');
var WorldInfo = require('./x3d.mjs');
var MFString = require('./x3d.mjs');
var Transform = require('./x3d.mjs');
var SFVec3f = require('./x3d.mjs');
var Anchor = require('./x3d.mjs');
var SFRotation = require('./x3d.mjs');
var Shape = require('./x3d.mjs');
var Appearance = require('./x3d.mjs');
var ImageTexture = require('./x3d.mjs');
var IndexedFaceSet = require('./x3d.mjs');
var MFInt32 = require('./x3d.mjs');
var TextureCoordinate = require('./x3d.mjs');
var MFVec2f = require('./x3d.mjs');
var Coordinate = require('./x3d.mjs');
var MFVec3f = require('./x3d.mjs');
var Material = require('./x3d.mjs');
var SFFloat = require('./x3d.mjs');
var SFColor = require('./x3d.mjs');
var Text = require('./x3d.mjs');
var MFFloat = require('./x3d.mjs');
var FontStyle = require('./x3d.mjs');
var X3D0 =  new X3D({

      profile : new SFString("Full"),
      version : new SFString("3.3"),
      head : new SFNode(
        new head({
          meta : new MFNode([
            new meta({
              name : new SFString("comment"),
              content : new SFString("World of Titania")}),

            new meta({
              name : new SFString("created"),
              content : new SFString("Thu, 23 Apr 2015 06:07:09 GMT")}),

            new meta({
              name : new SFString("creator"),
              content : new SFString("Holger Seelig")}),

            new meta({
              name : new SFString("generator"),
              content : new SFString("Titania V0.7.8, http://titania.create3000.de")}),

            new meta({
              name : new SFString("modified"),
              content : new SFString("Thu, 23 Apr 2015 06:07:09 GMT")})])})),
      Scene : new SFNode(
        new Scene({
          children : new MFNode([
            new WorldInfo({
              info : new MFString([", ","Created in TitaniaPackaged by CosmoPackage"])}),

            new Transform({
              DEF : new SFString("info"),
              translation : new SFVec3f([-0.0641994,0,0]),
              children : new MFNode([
                new Transform({
                  DEF : new SFString("link"),
                  children : new MFNode([
                    new Anchor({
                      description : new SFString("Klicken Sie hier fr eine Reservierung."),
                      url : new MFString(["javascript: function tmt_winLaunch(theURL,winName,targetName,features) { eval(winName+\"=window.open('\"+theURL+\"','\"+targetName+\"','\"+features+\"')\") } tmt_winLaunch('/3D/html/mbenz/placeholder.html', 'place','place','resizable=yes,width=623,height=454,left=0,top=0');"]),
                      children : new MFNode([
                        new Transform({}),

                        new Transform({
                          translation : new SFVec3f([14.9304,1.44228,-4.53928]),
                          rotation : new SFRotation([0,1,0,4.71239]),
                          scale : new SFVec3f([1.15231,0.684961,0.863441]),
                          children : new MFNode([
                            new Shape({
                              appearance : new SFNode(
                                new Appearance({
                                  texture : new SFNode(
                                    new ImageTexture({
                                      DEF : new SFString("_1"),
                                      url : new MFString(["reflect1.png"])}))})),
                              geometry : new SFNode(
                                new IndexedFaceSet({
                                  texCoordIndex : new MFInt32([4,2,3,5,-1,0,1,6,7,-1,8,9,10,11,-1]),
                                  coordIndex : new MFInt32([4,2,3,5,-1,0,1,6,7,-1,6,4,5,7,-1]),
                                  texCoord : new SFNode(
                                    new TextureCoordinate({
                                      point : new MFVec2f([0,0,1,0,1,1,0,1,1,0.5,0,0.5,1,0.25,0,0.25,0.271287,0.0518155,0.271287,0.418541,0.0469032,0.418541,0.0469032,0.0518155])})),
                                  coord : new SFNode(
                                    new Coordinate({
                                      point : new MFVec3f([-2,-0.487787,0,2,-0.487787,0,2,0.575824,0,-2,0.575824,0,2,0.480384,0,-2,0.480384,0,2,-0.390195,0,-2,-0.390195,0])}))}))})])}),

                        new Transform({
                          translation : new SFVec3f([14.8995,1.3026,-6.67413]),
                          rotation : new SFRotation([0,1,0,4.71239]),
                          scale : new SFVec3f([0.0311052,0.0311052,0.0311052]),
                          children : new MFNode([
                            new Shape({
                              appearance : new SFNode(
                                new Appearance({
                                  material : new SFNode(
                                    new Material({
                                      DEF : new SFString("_2"),
                                      ambientIntensity : new SFFloat(0),
                                      diffuseColor : new SFColor([0,0,0]),
                                      emissiveColor : new SFColor([0.0105023,0,0.255319])}))})),
                              geometry : new SFNode(
                                new Text({
                                  string : new MFString([", ","Jetzt reservieren! "]),
                                  length : new MFFloat([0]),
                                  fontStyle : new SFNode(
                                    new FontStyle({
                                      family : new MFString(["SANS"]),
                                      size : new SFFloat(18)}))}))})])})])})])}),

                new Transform({
                  DEF : new SFString("reservier"),
                  children : new MFNode([
                    new Transform({
                      translation : new SFVec3f([14.9236,2.66827,-5.48662]),
                      rotation : new SFRotation([0,-1,0,1.5708]),
                      scale : new SFVec3f([0.0341302,0.0341301,0.0341303]),
                      children : new MFNode([
                        new Shape({
                          appearance : new SFNode(
                            new Appearance({
                              material : new SFNode(
                                new Material({
                                  USE : new SFString("_2")}))})),
                          geometry : new SFNode(
                            new Text({
                              string : new MFString([", ","Mercedes-Benz lädt Sie, ","zu einer Probefahrt ein! "]),
                              length : new MFFloat([0]),
                              fontStyle : new SFNode(
                                new FontStyle({
                                  family : new MFString(["SANS"]),
                                  size : new SFFloat(14),
                                  justify : new MFString(["MIDDLE"])}))}))})])})])}),

                new Transform({
                  DEF : new SFString("slogan"),
                  translation : new SFVec3f([14.9259,3.48694,-7.03026]),
                  rotation : new SFRotation([0,-1,0,1.5708]),
                  scale : new SFVec3f([0.0311052,0.0311052,0.0311052]),
                  children : new MFNode([
                    new Shape({
                      appearance : new SFNode(
                        new Appearance({
                          material : new SFNode(
                            new Material({
                              USE : new SFString("_2")}))})),
                      geometry : new SFNode(
                        new Text({
                          string : new MFString(["neues Design, neue Ausstattung, mehr Spaß"]),
                          length : new MFFloat([0]),
                          fontStyle : new SFNode(
                            new FontStyle({
                              family : new MFString(["SANS"]),
                              size : new SFFloat(18)}))}))})])}),

                new Transform({
                  DEF : new SFString("title"),
                  translation : new SFVec3f([14.9259,4.16814,-8.31291]),
                  rotation : new SFRotation([0,-1,0,1.5708]),
                  scale : new SFVec3f([0.0311052,0.0311052,0.0311052]),
                  children : new MFNode([
                    new Shape({
                      appearance : new SFNode(
                        new Appearance({
                          material : new SFNode(
                            new Material({
                              USE : new SFString("_2")}))})),
                      geometry : new SFNode(
                        new Text({
                          string : new MFString(["Die neue A-Klasse ist da!"]),
                          length : new MFFloat([0]),
                          fontStyle : new SFNode(
                            new FontStyle({
                              family : new MFString(["SANS"]),
                              size : new SFFloat(24)}))}))})])}),

                new Transform({
                  translation : new SFVec3f([14.9304,4.4196,-3.98687]),
                  rotation : new SFRotation([0,1,0,4.71239]),
                  scale : new SFVec3f([2.29482,0.793292,1]),
                  children : new MFNode([
                    new Shape({
                      appearance : new SFNode(
                        new Appearance({
                          texture : new SFNode(
                            new ImageTexture({
                              USE : new SFString("_1")}))})),
                      geometry : new SFNode(
                        new IndexedFaceSet({
                          texCoordIndex : new MFInt32([4,2,3,5,-1,0,1,6,7,-1,8,9,10,11,-1]),
                          coordIndex : new MFInt32([4,2,3,5,-1,0,1,6,7,-1,6,4,5,7,-1]),
                          texCoord : new SFNode(
                            new TextureCoordinate({
                              point : new MFVec2f([0,0,1,0,1,1,0,1,1,0.5,0,0.5,1,0.25,0,0.25,0.271287,0.0518155,0.271287,0.418541,0.0469032,0.418541,0.0469032,0.0518155])})),
                          coord : new SFNode(
                            new Coordinate({
                              point : new MFVec3f([-2,-0.487787,0,2,-0.487787,0,2,0.575824,0,-2,0.575824,0,2,0.480384,0,-2,0.480384,0,2,-0.390195,0,-2,-0.390195,0])}))}))})])})])})])}))});
console.log(X3D0.toXMLNode());

File: src/main/c/net/coderextreme/Library/Examples/ICTS/stage/infowall2.c
#include "C:/x3d-code/www.web3d.org/x3d/languages/c/Concretes.h"
void main(int argc, char ** argv) {
Browser browser = X3D.getBrowser();
X3D X3D0;
X3D0.profile = "Full";
X3D0.version = "3.3";
head head1 = createNode("head");
meta meta2 = createNode("meta");
meta2.name = "comment";
meta2.content = "World of Titania";
head1.meta = new MFNode();

head1.meta[0] = meta2;

meta meta3 = createNode("meta");
meta3.name = "created";
meta3.content = "Thu, 23 Apr 2015 06:07:09 GMT";
head1.meta[1] = meta3;

meta meta4 = createNode("meta");
meta4.name = "creator";
meta4.content = "Holger Seelig";
head1.meta[2] = meta4;

meta meta5 = createNode("meta");
meta5.name = "generator";
meta5.content = "Titania V0.7.8, http://titania.create3000.de";
head1.meta[3] = meta5;

meta meta6 = createNode("meta");
meta6.name = "modified";
meta6.content = "Thu, 23 Apr 2015 06:07:09 GMT";
head1.meta[4] = meta6;

head = head1;

WorldInfo WorldInfo8 = createNode("WorldInfo");
WorldInfo8.info = new MFString(new java.lang.String[", ","Created in TitaniaPackaged by CosmoPackage"]);
children = new MFNode();

children[0] = WorldInfo8;

Transform Transform9 = createNode("Transform");
Transform9.DEF = "info";
Transform9.translation = new SFVec3f(new float[-0.0641994,0,0]);
Transform Transform10 = createNode("Transform");
Transform10.DEF = "link";
Anchor Anchor11 = createNode("Anchor");
Anchor11.description = "Klicken Sie hier fr eine Reservierung.";
Anchor11.url = new MFString(new java.lang.String["javascript: function tmt_winLaunch(theURL,winName,targetName,features) { eval(winName+\"=window.open('\"+theURL+\"','\"+targetName+\"','\"+features+\"')\") } tmt_winLaunch('/3D/html/mbenz/placeholder.html', 'place','place','resizable=yes,width=623,height=454,left=0,top=0');"]);
Transform Transform12 = createNode("Transform");
Anchor11.children = new MFNode();

Anchor11.children[0] = Transform12;

Transform Transform13 = createNode("Transform");
Transform13.translation = new SFVec3f(new float[14.9304,1.44228,-4.53928]);
Transform13.rotation = new SFRotation(new float[0,1,0,4.71239]);
Transform13.scale = new SFVec3f(new float[1.15231,0.684961,0.863441]);
Shape Shape14 = createNode("Shape");
Appearance Appearance15 = createNode("Appearance");
ImageTexture ImageTexture16 = createNode("ImageTexture");
ImageTexture16.DEF = "_1";
ImageTexture16.url = new MFString(new java.lang.String["reflect1.png"]);
Appearance15.texture = ImageTexture16;

Shape14.appearance = Appearance15;

IndexedFaceSet IndexedFaceSet17 = createNode("IndexedFaceSet");
IndexedFaceSet17.texCoordIndex = new MFInt32(new int[4,2,3,5,-1,0,1,6,7,-1,8,9,10,11,-1]);
IndexedFaceSet17.coordIndex = new MFInt32(new int[4,2,3,5,-1,0,1,6,7,-1,6,4,5,7,-1]);
TextureCoordinate TextureCoordinate18 = createNode("TextureCoordinate");
TextureCoordinate18.point = new MFVec2f(new float[0,0,1,0,1,1,0,1,1,0.5,0,0.5,1,0.25,0,0.25,0.271287,0.0518155,0.271287,0.418541,0.0469032,0.418541,0.0469032,0.0518155]);
IndexedFaceSet17.texCoord = TextureCoordinate18;

Coordinate Coordinate19 = createNode("Coordinate");
Coordinate19.point = new MFVec3f(new float[-2,-0.487787,0,2,-0.487787,0,2,0.575824,0,-2,0.575824,0,2,0.480384,0,-2,0.480384,0,2,-0.390195,0,-2,-0.390195,0]);
IndexedFaceSet17.coord = Coordinate19;

Shape14.geometry = IndexedFaceSet17;

Transform13.child = new undefined();

Transform13.child[0] = Shape14;

Anchor11.children[1] = Transform13;

Transform Transform20 = createNode("Transform");
Transform20.translation = new SFVec3f(new float[14.8995,1.3026,-6.67413]);
Transform20.rotation = new SFRotation(new float[0,1,0,4.71239]);
Transform20.scale = new SFVec3f(new float[0.0311052,0.0311052,0.0311052]);
Shape Shape21 = createNode("Shape");
Appearance Appearance22 = createNode("Appearance");
Material Material23 = createNode("Material");
Material23.DEF = "_2";
Material23.ambientIntensity = 0;
Material23.diffuseColor = new SFColor(new float[0,0,0]);
Material23.emissiveColor = new SFColor(new float[0.0105023,0,0.255319]);
Appearance22.material = Material23;

Shape21.appearance = Appearance22;

Text Text24 = createNode("Text");
Text24.string = new MFString(new java.lang.String[", ","Jetzt reservieren! "]);
Text24.length = new MFFloat(new float[0]);
FontStyle FontStyle25 = createNode("FontStyle");
FontStyle25.family = new MFString(new java.lang.String["SANS"]);
FontStyle25.size = 18;
Text24.fontStyle = FontStyle25;

Shape21.geometry = Text24;

Transform20.child = new undefined();

Transform20.child[0] = Shape21;

Anchor11.children[2] = Transform20;

Transform10.children = new MFNode();

Transform10.children[0] = Anchor11;

Transform9.children = new MFNode();

Transform9.children[0] = Transform10;

Transform Transform26 = createNode("Transform");
Transform26.DEF = "reservier";
Transform Transform27 = createNode("Transform");
Transform27.translation = new SFVec3f(new float[14.9236,2.66827,-5.48662]);
Transform27.rotation = new SFRotation(new float[0,-1,0,1.5708]);
Transform27.scale = new SFVec3f(new float[0.0341302,0.0341301,0.0341303]);
Shape Shape28 = createNode("Shape");
Appearance Appearance29 = createNode("Appearance");
Material Material30 = createNode("Material");
Material30.USE = "_2";
Appearance29.material = Material30;

Shape28.appearance = Appearance29;

Text Text31 = createNode("Text");
Text31.string = new MFString(new java.lang.String[", ","Mercedes-Benz lädt Sie, ","zu einer Probefahrt ein! "]);
Text31.length = new MFFloat(new float[0]);
FontStyle FontStyle32 = createNode("FontStyle");
FontStyle32.family = new MFString(new java.lang.String["SANS"]);
FontStyle32.size = 14;
FontStyle32.justify = new MFString(new java.lang.String["MIDDLE"]);
Text31.fontStyle = FontStyle32;

Shape28.geometry = Text31;

Transform27.child = new undefined();

Transform27.child[0] = Shape28;

Transform26.children = new MFNode();

Transform26.children[0] = Transform27;

Transform9.children[1] = Transform26;

Transform Transform33 = createNode("Transform");
Transform33.DEF = "slogan";
Transform33.translation = new SFVec3f(new float[14.9259,3.48694,-7.03026]);
Transform33.rotation = new SFRotation(new float[0,-1,0,1.5708]);
Transform33.scale = new SFVec3f(new float[0.0311052,0.0311052,0.0311052]);
Shape Shape34 = createNode("Shape");
Appearance Appearance35 = createNode("Appearance");
Material Material36 = createNode("Material");
Material36.USE = "_2";
Appearance35.material = Material36;

Shape34.appearance = Appearance35;

Text Text37 = createNode("Text");
Text37.string = new MFString(new java.lang.String["neues Design, neue Ausstattung, mehr Spaß"]);
Text37.length = new MFFloat(new float[0]);
FontStyle FontStyle38 = createNode("FontStyle");
FontStyle38.family = new MFString(new java.lang.String["SANS"]);
FontStyle38.size = 18;
Text37.fontStyle = FontStyle38;

Shape34.geometry = Text37;

Transform33.child = new undefined();

Transform33.child[0] = Shape34;

Transform9.children[2] = Transform33;

Transform Transform39 = createNode("Transform");
Transform39.DEF = "title";
Transform39.translation = new SFVec3f(new float[14.9259,4.16814,-8.31291]);
Transform39.rotation = new SFRotation(new float[0,-1,0,1.5708]);
Transform39.scale = new SFVec3f(new float[0.0311052,0.0311052,0.0311052]);
Shape Shape40 = createNode("Shape");
Appearance Appearance41 = createNode("Appearance");
Material Material42 = createNode("Material");
Material42.USE = "_2";
Appearance41.material = Material42;

Shape40.appearance = Appearance41;

Text Text43 = createNode("Text");
Text43.string = new MFString(new java.lang.String["Die neue A-Klasse ist da!"]);
Text43.length = new MFFloat(new float[0]);
FontStyle FontStyle44 = createNode("FontStyle");
FontStyle44.family = new MFString(new java.lang.String["SANS"]);
FontStyle44.size = 24;
Text43.fontStyle = FontStyle44;

Shape40.geometry = Text43;

Transform39.child = new undefined();

Transform39.child[0] = Shape40;

Transform9.children[3] = Transform39;

Transform Transform45 = createNode("Transform");
Transform45.translation = new SFVec3f(new float[14.9304,4.4196,-3.98687]);
Transform45.rotation = new SFRotation(new float[0,1,0,4.71239]);
Transform45.scale = new SFVec3f(new float[2.29482,0.793292,1]);
Shape Shape46 = createNode("Shape");
Appearance Appearance47 = createNode("Appearance");
ImageTexture ImageTexture48 = createNode("ImageTexture");
ImageTexture48.USE = "_1";
Appearance47.texture = ImageTexture48;

Shape46.appearance = Appearance47;

IndexedFaceSet IndexedFaceSet49 = createNode("IndexedFaceSet");
IndexedFaceSet49.texCoordIndex = new MFInt32(new int[4,2,3,5,-1,0,1,6,7,-1,8,9,10,11,-1]);
IndexedFaceSet49.coordIndex = new MFInt32(new int[4,2,3,5,-1,0,1,6,7,-1,6,4,5,7,-1]);
TextureCoordinate TextureCoordinate50 = createNode("TextureCoordinate");
TextureCoordinate50.point = new MFVec2f(new float[0,0,1,0,1,1,0,1,1,0.5,0,0.5,1,0.25,0,0.25,0.271287,0.0518155,0.271287,0.418541,0.0469032,0.418541,0.0469032,0.0518155]);
IndexedFaceSet49.texCoord = TextureCoordinate50;

Coordinate Coordinate51 = createNode("Coordinate");
Coordinate51.point = new MFVec3f(new float[-2,-0.487787,0,2,-0.487787,0,2,0.575824,0,-2,0.575824,0,2,0.480384,0,-2,0.480384,0,2,-0.390195,0,-2,-0.390195,0]);
IndexedFaceSet49.coord = Coordinate51;

Shape46.geometry = IndexedFaceSet49;

Transform45.child = new undefined();

Transform45.child[0] = Shape46;

Transform9.children[4] = Transform45;

children[1] = Transform9;

}
